server/service/src/repack/validate.rs

```diff
@@ -0,0 +1,65 @@
+use repository::{
+    EqualFilter, RepositoryError, StockLine, StockLineFilter, StockLineRepository, StockLineRow,
+    StorageConnection,
+};
+
+use super::{
+    common::calculate_stock_line_total,
+    insert::{InsertRepack, InsertRepackError},
+};
+
+pub fn validate(
+    connection: &StorageConnection,
+    store_id: &str,
+    input: &InsertRepack,
+) -> Result<StockLineRow, InsertRepackError> {
+    let stock_line = match check_stock_line_exists(connection, store_id, &input.stock_line_id)? {
```

Comment:
hmm, we have a few lines like this, can they be consolidated ? (it's a nit: doesn't need to be done in this PR, but nice to do).

If i was to consolidate it, I would return three types of error and map them:
```rust
let stock_line = check_stock_line_exists(con, store_id, stock_line_id).map_err(|err| => match err {
   NotFound => StockLineDoesNotExist,
   NotThisStore => NotThisStoreStockLine,
   RespositoryError => DatabaseError
})?;


```

Reply:
Made all the changes 🙏🏻